openhands/llm/llm.py

```diff
@@ -327,6 +330,18 @@ def is_caching_prompt_active(self) -> bool:
             )
         )
 
+    def is_function_calling_active(self) -> bool:
+        # Check if model name is in supported list before checking model_info
+        model_name_supported = (
+            self.config.model in FUNCTION_CALLING_SUPPORTED_MODELS
+            or self.config.model.split('/')[-1] in FUNCTION_CALLING_SUPPORTED_MODELS
+            or any(m in self.config.model for m in FUNCTION_CALLING_SUPPORTED_MODELS)
+        )
+        return model_name_supported or (
+            self.model_info is not None
+            and self.model_info.get('supports_function_calling', False)
```

Comment:
This way though, it will return `active` for all models in litellm's list. That includes DeepSeek, doesn't it?

Reply:
oh shoo you caught me - not enough coffee today - will fix!

Reply:
I think we will need to change it back to "and" then - we will need model_info to be None, which will hold true for most case (except litellm proxy using `openai/` as prefix).

Reply:
Why would we want people to switch to litellm proxy?

Reply:
I mean, most users, as far as I know, don't use litellm proxy at all. They use directly Anthropic / OpenAI / etc APIs, they use providers like bedrock or groq etc. I don't know, not many install the litellm proxy.

Many (most?) of those APIs are in litellm's `model info`. So it would return a result about function calling, just like for Deepseek (via https://api.deepseek.com/v1) it returns True. 🤔

Reply:
> They use directly Anthropic / OpenAI / etc APIs, they use providers like bedrock or groq etc.

Yeah - i think this case, the model info will exists, so "model_name_supported and (..)" will work.

I think the purpose here is just to return False for deepseek, which will also work - since model_name_supported will be False in this case.

Reply:
I think you're right... 🤔 It should work. You're not the only one in need of coffee!